tests/system/tokenizer/test_cohere_api_tokenizer.py

```diff
@@ -1,91 +1,55 @@
+               "spaces   \n \n\n CASE cAse "
 
     @staticmethod
     @pytest.fixture
     def expected_tokens(text):
-        return ["Hello", " World", "!"]
-
-    # region: test tokenize
-
-    @staticmethod
-    def test_tokenize(tokenizer, text, expected_tokens):
-        tokens = tokenizer.tokenize(text)
-        assert tokens == expected_tokens, f"\nExpected: {expected_tokens}" \
-                                          f"\nActual: {tokens}"
-
-    @staticmethod
-    def test_tokenize_empty_string(tokenizer):
-        assert tokenizer.tokenize("") == []
-
-    @staticmethod
-    def test_tokenize_invalid_input_type_raise_exception(tokenizer):
-        with pytest.raises(Exception):
-            tokenizer.tokenize(1)
-
-        with pytest.raises(Exception):
-            tokenizer.tokenize(["asd"])
-
-    # endregion
-
-    # region: test detokenize
+        return ['string', ' with', ' special', ' characters', ' like',
+                ' !', '@', '#', '$', '%', '^', '&', '*', '()', '_', '+', '日',
+                '本', ' spaces', '   ', '\n ', '\n\n', ' CASE', ' c', 'A',
+                'se', " "]
 
     @staticmethod
-    def test_detokenize(tokenizer, text, expected_tokens):
-        text = tokenizer.detokenize(expected_tokens)
-        assert text == text
+    def test_messages_token_count(tokenizer):
+        messages = [MessageBase(role=Role.USER, content="Hello, assistant.")]
+        assert tokenizer.messages_token_count(messages) == 11
 
-    @staticmethod
-    def test_detokenize_empty_string(tokenizer):
-        assert tokenizer.detokenize([]) == ""
-
-    @staticmethod
-    def test_detokenize_invalid_input_type_raise_exception(tokenizer):
-        with pytest.raises(Exception):
-            tokenizer.detokenize(1)
-
-        with pytest.raises(Exception):
-            tokenizer.detokenize("asd")
-
-    # region test token_count
-
-    @staticmethod
-    def test_token_count(tokenizer, text, expected_tokens):
-        token_count = tokenizer.token_count(text)
-        assert token_count == len(expected_tokens)
-        assert token_count == len(tokenizer.tokenize(text))
-
-    @staticmethod
-    def test_token_count_empty_string(tokenizer):
-        assert tokenizer.token_count("") == 0
+        messages = [MessageBase(role=Role.USER,
+                                content="Hello, assistant."),
+                    MessageBase(role=Role.ASSISTANT,
+                                content="Hello, user. How can I assist you?")]
+        assert tokenizer.messages_token_count(messages) == 25
 
     @staticmethod
     def test_messages_token_count_empty_messages(tokenizer):
         assert tokenizer.messages_token_count([]) == 3
 
-    # endregion
-
-    # region special tokens
-
     @staticmethod
     def test_special_tokens_to_natural_text(tokenizer):
-        tokens = tokenizer.tokenize("<|endoftext|>")
+        tokens = tokenizer.tokenize("<BOS_TOKEN>")
```

Comment:
@scottmx81 Please note that this test still fails - for both `test_cohere_api_tokenizer` and `test_cohere_hf_tokenizer`.  
The expected behavior is that the string `'<BOS_TOKEN>'` is treated as a string, not an actual token (as it may naturally occur in the user's text).  

Can you please work with the Cohere people to see if there's a way to configure "ignore token-like strings"?

Reply:
@igiloh-pinecone I have looked into this today and raised the issue, and it turns out that Cohere does not support having special token symbols in the prompt. This is the expected behaviour from the Cohere tokenizers right now. These special token symbols in the prompts will effectively be dropped by the tokenizers.

I know that there are edge cases where a user may want to do this, such as if the user were to ask the chatbot what that token symbol means. This is an edge case that Cohere does not support right now.

Therefore I think it is best to remove this test. To the best of my knowledge, there is no escape function or "ignore token-like strings" flag. The count being returned is the real and expected count, at least as far as the Cohere tokenizers are concerned.

Reply:
Understood. I agree this is an edge case, we can skip the test or simply delete it.